src/test/mir-opt/inline/inline_instruction_set.rs

```diff
@@ -41,8 +43,6 @@ fn instruction_set_default() {}
 pub fn t32() {
     instruction_set_a32();
     instruction_set_t32();
-    // The default instruction set is currently
-    // conservatively assumed to be incompatible.
     instruction_set_default();
```

Comment:
Could you add a test case demonstrating the check you added on `InlineAsm` terminators?

Reply:
I'm unclear on what that would involve because I'm not really sure what the tests are doing as is, since they don't appear to assert anything.

Do I just add a function that uses an inline assembly block?

Reply:
Currently there is only one non-specific assertion about content of the whole MIR - and MIR needs to be reviewed manually when it does change.

Calling a function containing inline assembly that doesn't have the instruction set attribute (and has the inline always attribute, to avoid doubts about threshold) from a function that does have the attribute sounds right.

Reply:
How do I insert real inline assembly into the test? At the moment the test file is defining the `asm!` macro itself, and just defining it as a dummy macro. Unless maybe that definition is "fake" and the `asm!` macro always exists?

Reply:
The `rustc_builtin_macro` attribute should mark it as builtin macro. Did you check if it just works as is?

Reply:
It _does_ build, but that by itself doesn't make me trust it because I don't trust anything at all to do with this strange test file.

Reply:
To test the inline assembly check, the callee with an inline assembly should omit the `instruction_set` attribute.

Reply:
Ah, can do.